src/db/mod.rs

```diff
@@ -56,230 +51,35 @@ pub struct Database {
-
-    #[error("Incomplete outer header: Missing {}", missing_field)]
-    IncompleteOuterHeader { missing_field: String },
-
-    #[error("Invalid inner header entry: {}", entry_type)]
-    InvalidInnerHeaderEntry { entry_type: u8 },
-
-    #[error("Incomplete outer header: Missing {}", missing_field)]
-    IncompleteInnerHeader { missing_field: String },
-
-    #[error(transparent)]
-    Cryptography(#[from] CryptographyError),
-
-    #[error(transparent)]
-    Xml(#[from] XmlParseError),
-
-    #[error(transparent)]
-    OuterCipher(#[from] OuterCipherSuiteError),
-
-    #[error(transparent)]
-    InnerCipher(#[from] InnerCipherSuiteError),
-
-    #[error(transparent)]
-    Compression(#[from] CompressionError),
-
-    #[error(transparent)]
-    BlockStream(#[from] BlockStreamError),
-
-    #[error(transparent)]
-    VariantDictionary(#[from] VariantDictionaryError),
-
-    #[error(transparent)]
-    KdfSettings(#[from] KdfSettingsError),
-
-    #[error(transparent)]
-    Io(#[from] std::io::Error),
-}
-
-#[derive(Debug, Error)]
-pub enum DatabaseSaveError {
-    #[error("Saving this database version is not supported")]
-    UnsupportedVersion,
-
-    #[error("Error while generating XML")]
-    Xml(#[from] xml::writer::Error),
-
-    #[error(transparent)]
-    Io(#[from] std::io::Error),
-
-    #[error(transparent)]
-    Key(#[from] DatabaseKeyError),
-
-    #[error(transparent)]
-    Cryptography(#[from] CryptographyError),
-
-    #[error(transparent)]
-    Random(#[from] getrandom::Error),
-}
-
-impl From<CryptographyError> for DatabaseOpenError {
-    fn from(e: CryptographyError) -> Self {
-        DatabaseIntegrityError::from(e).into()
-    }
-}
-
-impl From<BlockStreamError> for DatabaseOpenError {
-    fn from(e: BlockStreamError) -> Self {
-        DatabaseIntegrityError::from(e).into()
-    }
-}
-
-impl From<XmlParseError> for DatabaseOpenError {
-    fn from(e: XmlParseError) -> Self {
-        DatabaseIntegrityError::from(e).into()
-    }
-}
-
-impl From<InnerCipherSuiteError> for DatabaseOpenError {
-    fn from(e: InnerCipherSuiteError) -> Self {
-        DatabaseIntegrityError::from(e).into()
-    }
-}
-
-impl From<OuterCipherSuiteError> for DatabaseOpenError {
-    fn from(e: OuterCipherSuiteError) -> Self {
-        DatabaseIntegrityError::from(e).into()
-    }
-}
-
-impl From<KdfSettingsError> for DatabaseOpenError {
-    fn from(e: KdfSettingsError) -> Self {
-        DatabaseIntegrityError::from(e).into()
-    }
-}
-
-impl From<VariantDictionaryError> for DatabaseOpenError {
-    fn from(e: VariantDictionaryError) -> Self {
-        DatabaseIntegrityError::from(e).into()
-    }
-}
-
-impl From<CompressionError> for DatabaseOpenError {
-    fn from(e: CompressionError) -> Self {
-        DatabaseIntegrityError::from(e).into()
-    }
-}
-
+/// Settings for how a database is stored on disk
 #[derive(Debug)]
 #[cfg_attr(feature = "serialization", derive(serde::Serialize))]
 pub struct DatabaseSettings {
```

Comment:
@sseemayer should that not be `DatabaseConfig`? To be consistent with the other configuration structs.

Reply:
Good point, I've renamed this and a bunch of other types to have a consistent naming.